### PR TITLE
40 us as a chat user i want to invite other users to play pong game

### PR DESCRIPTION
**Pull request by COTHSC**
# Description

Please provide a detailed description of what was done in this PR.
Precise the issue that you are resolving.

# Changes include

- [ ] New feature (non-breaking change that adds functionality)
- [ ] Bugfix (non-breaking change that solves an issue)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it.

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the needed labels
- [x] I have linked this PR to an issue
- [x] I have linked this PR to a milestone
- [x] I have linked this PR to a project
- [ ] I have tested this code
- [ ] I have added / updated tests (unit / functionals / end-to-end / ...)
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it.

### CODE REVIEW

File: backend/app/src/pong/entities/game.entities.ts [hunk 1]
```diff
@@ -7,6 +7,21 @@ export interface HandshakeRequest extends Request {
...
+  PLAYING = 'PLAYING',
+}
+
+export interface FrontendUser {
+  id: string;
+  avatarImg: string;
+  nickname: string;
+  eloScore: number;
+  twoFactorAuthenticationSet: boolean;
+  twoFactorAuthenticationSecret: string;
```
Comment (llescure) [inline backend/app/src/pong/entities/game.entities.ts:22]: We deleted the secret for security reason

File: backend/app/src/pong/game.service.ts [hunk 2]
```diff
@@ -31,6 +33,51 @@ export class GameService {
...
+        id: p1id,
+      },
+      select: {
+        id: true,
+        avatarImg: true,
+        nickname: true,
+        eloScore: true,
+        status: true,
+        twoFactorAuthenticationSet: true,
+        twoFactorAuthenticationSecret: true,
```
Comment (llescure) [inline backend/app/src/pong/game.service.ts:72]: Don't send the Secret!

File: backend/app/src/user/user.service.ts [hunk 3]
```diff
@@ -128,10 +128,11 @@ export class UserService {
   }
 
   async getUserInfo(userId: string): Promise<User | undefined> {
-    const user: User = await this.prismaService.user.findUnique({
+    const user = await this.prismaService.user.findUnique({
       where: {
         id: userId,
       },
+      //TODO select useful info
```
Comment (llescure) [inline backend/app/src/user/user.service.ts:135]: Done in a pending PR !

**Review by llescure — COMMENTED** [review]
Add a timeout for invitation

**Review by llescure — COMMENTED** [review]
Inviting a user to play in the channel section does not end well ...

File: backend/app/src/pong/game.gateway.ts [hunk 4]
```diff
@@ -58,6 +58,33 @@ export class GameGateway {
     return this.gameService.rejoin(userId);
   }
 
+  @UseGuards(JwtAuthGuard)
+  @SubscribeMessage('refuseInvite')
+  refuseGameInvite(
+    @Body() challenger: FrontendUser,
+    @ConnectedSocket() client: Socket,
+  ) {
+    this.gameService.cancelInvite(client, challenger.id);
```
[the remark this answers is not in the document]
Comment (cnstll) [reply backend/app/src/pong/game.gateway.ts:66]: renamed

File: backend/app/src/pong/game.gateway.ts [hunk 5]
```diff
@@ -19,6 +19,7 @@ import { GameMode } from './entities/game.entities';
       process.env.BACKEND_URL,
       process.env.DOMAIN,
       process.env.PUBLIC_URL,
+      'http://localhost',
```
[the remark this answers is not in the document]
Comment (cnstll) [reply backend/app/src/pong/game.gateway.ts]: removed

File: backend/app/src/pong/entities/game.entities.ts (hunk 1 above)
Comment (cnstll) [reply backend/app/src/pong/entities/game.entities.ts:22]: removed

File: backend/app/src/pong/game.service.ts (hunk 2 above)
Comment (cnstll) [reply backend/app/src/pong/game.service.ts]: removed

File: backend/app/src/user/user.service.ts (hunk 3 above)
Comment (cnstll) [reply backend/app/src/user/user.service.ts:135]: cheers

**Comment by cnstll** [comment]
@llescure reviewed fixed, good for testing and merge

**Review by llescure — COMMENTED** [review]
Screenshot 2022-11-24 at 11 11 44
Following error when a user has an invitation timeout and accept or refuse

**Review by llescure — COMMENTED** [review]
Invite with a seed user doesn't work, "invitation refused", we have the following log
Screenshot 2022-11-24 at 11 21 14

**Comment by cnstll** [comment]
@llescure 
- no timeout error now handled with try-catch blocks
- invite between seed users now working

File: backend/app/src/pong/game.service.ts [hunk 6]
```diff
@@ -96,6 +171,25 @@ export class GameService {
...
+    challengerSocketId: string,
+    opponentSocketId: string,
+    challengerId: string,
+  ) {
+    const callback = () => {
+      this.GameMap.delete(challengerId);
+      server
+        .to(challengerSocketId)
+        .emit('inviteRefused', 'Your opponent is to slow for you');
+      server.to(opponentSocketId).emit('inviteRefused', 'YOU were to slow');
```
Comment (llescure) [inline backend/app/src/pong/game.service.ts:186]: YOU were too slow

File: backend/app/src/pong/game.service.ts [hunk 7]
```diff
@@ -96,6 +171,25 @@ export class GameService {
...
+    server: Server,
+    challengerSocketId: string,
+    opponentSocketId: string,
+    challengerId: string,
+  ) {
+    const callback = () => {
+      this.GameMap.delete(challengerId);
+      server
+        .to(challengerSocketId)
+        .emit('inviteRefused', 'Your opponent is to slow for you');
```
Comment (llescure) [inline backend/app/src/pong/game.service.ts:185]: Your opponent is too slow for you

**Review by llescure — APPROVED** [review]

**Pull request merged**